lib/acme/certificate.rb

```diff
@@ -9,6 +9,46 @@ def initialize(certificate, chain, request)
     @x509 = certificate
     @x509_chain = chain
     @request = request
+
+    #create_self_signed_cert if x509.nil?
+  end
+
+  def create_self_signed_cert subjectAltDomains
+    keypair = OpenSSL::PKey::RSA.new(2048) 
```

Comment:
I disagree, private keys usually have a higher protection than the corresponding certificate. It does seem contrived, but being able to read but not write an existing private key that must be used is not unimaginable for me. Besides the contrived real world scenario, being able to provide an existing private key will speed up the unit tests for this method by an order of magnitude easily.